Formula/kakoune.rb

```diff
@@ -21,19 +21,24 @@ class Kakoune < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "606496e883548440a2813b00e3639212da5d3aa11a52d4fd9ae180d373197fc5"
   end
 
-  depends_on macos: :high_sierra # needs C++17
-  depends_on "ncurses"
-
-  uses_from_macos "libxslt" => :build
+  on_macos do
+    depends_on "llvm" => :build if DevelopmentTools.clang_build_version <= 1200
+  end
```

Comment:
I think this would work too.
```suggestion
  uses_from_macos "llvm" => :build, since: :big_sur
```

Reply:
Doesn't this assume latest Xcode is installed which may not be the case? From what I can tell, [Big Sur support Xcodes < v12.5](https://developer.apple.com/support/xcode/) which wouldn't have a recent enough `clang`.

Reply:
If you don't have the latest version of brew dependencies you'll have a bad time either way. But you can add `depends_on "xcode" => "12.5"` if I remember the format correctly.

Reply:
Aren't these slightly different though (sorry for all the noob questions 🙈 - just trying to understand better).

I interpret:
- ```ruby
    on_macos do
      depends_on "llvm" => :build if DevelopmentTools.clang_build_version <= 1200
    end
  ```
  as: "install `llvm` if the system `clang` build number is <= 1200, thus allowing this formula to maybe potentially if you're lucky build on older systems"
- ```ruby
  depends_on "xcode" => "12.5"
  ```
  as: "depends on Xcode >= 12.5 to build, so if that is not supported by your OS, then tough luck"
  
With that in mind, I think `uses_from_macos "llvm" => :build, since: :big_sur` is a better approximation (also, confirmed this is the [correct syntax](https://github.com/Homebrew/homebrew-core/blob/7e80272b585ced37b11125d7c04ce77a44e00a76/Formula/doxygen.rb#LL28C3-L28C52)). Will change to this because it is simpler (might be worth also changing this in the linked examples above: [1](https://github.com/Homebrew/homebrew-core/blob/f8ba616a4ef7661ff2116414bc1973dce49279f5/Formula/vte3.rb#L44-L47) [2](https://github.com/Homebrew/homebrew-core/blob/f8ba616a4ef7661ff2116414bc1973dce49279f5/Formula/mold.rb#L33-L36) [3](https://github.com/Homebrew/homebrew-core/blob/f8ba616a4ef7661ff2116414bc1973dce49279f5/Formula/grokj2k.rb#L42-L45)).

Reply:
Implemented suggestion in 69187df5110272c31281ffb0a8ac02f84d83deb8

I also re-ran `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source kakoune` and the test, both worked.